### PR TITLE
Add support for newer ruby syntax

### PR DESCRIPTION
**Pull request by mzazrivec**

### CODE REVIEW

File: lib/gettext_i18n_rails/ruby_gettext_extractor.rb [hunk 1]
```diff
@@ -127,13 +122,22 @@ def new_call recv, meth, args = nil
...
+      Ruby20Parser
+    elsif RUBY_VERSION =~ /^2\.1/
+      Ruby21Parser
+    else
+      Ruby22Parser
+    end
   end
 
-  class Extractor19 < Ruby19Parser
+  class Extractor < get_ruby_parser
```
Comment (grosser) [inline lib/gettext_i18n_rails/ruby_gettext_extractor.rb]: ... inheriting from a method :D 
can you assign a constant like `BaseParser` or so instead ?

**Comment by grosser** [comment]
I like the look of this, but travis seems to be not that happy:

`uninitialized constant RubyGettextExtractor::Ruby22Parser`

File: Gemfile.lock [hunk 2]
```diff
@@ -7,125 +7,128 @@ PATH
 GEM
   remote: https://rubygems.org/
   specs:
-    actionmailer (4.2.0)
-      actionpack (= 4.2.0)
-      actionview (= 4.2.0)
-      activejob (= 4.2.0)
+    actionmailer (4.2.4)
```
Comment (grosser) [inline Gemfile.lock]: add a rails 4.2 gemfile too then :)

Comment (grosser) [inline Gemfile.lock]: (add in gemfiles and in .travis.yml)

**Comment by mzazrivec** [comment]
@grosser I think I was too fast submitting this pull request, since things do not seem to
work correctly with ruby 2.2.2 and rails 4.2 (which is what I meant to achieve).

To explain my motivation -- we use gettext_i18n_rails gem in ManageIQ application.
Recently, we started to use ruby 2.2 and rails 4.2 and with that, the gettext:find rake task
started to give this error:

```
$ bundle exec rake gettext:find
#<GetText::RubyLexX:0x0055dff10d8398>#<GetText::RubyLexX:0x0055dff10d8398>file ignored: ruby_parser cannot read haml files with 1.9 syntax --- app/views/layouts/_show_statistics.html.haml: ((string):20 :: parse error on value "%i(" (error))
...
```

Now when I'm testing my changes to the gettext_i18n_rails from this pull request (with ruby 2.2),
I'm getting the following error:

```
$ bundle exec rake --trace gettext:find
...
#<GetText::RubyLexX:0x0055de3ce3d558>#<GetText::RubyLexX:0x0055de3ce3d558>Error parsing app/views/ops/_db_info.html.haml
rake aborted!
NoMethodError: undefined method `size' for nil:NilClass
~/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/gettext_i18n_rails-1.2.3/lib/gettext_i18n_rails/ruby_gettext_extractor.rb:69:in `extract_key'
~/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/gettext_i18n_rails-1.2.3/lib/gettext_i18n_rails/ruby_gettext_extractor.rb:101:in `new_call'
~/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/ruby_parser-3.7.1/lib/ruby22_parser.rb:3773:in `_reduce_57'
~/.rbenv/versions/2.2.2/lib/ruby/2.2.0/racc/parser.rb:258:in `_racc_do_parse_c'
~/.rbenv/versions/2.2.2/lib/ruby/2.2.0/racc/parser.rb:258:in `do_parse'
~/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/ruby_parser-3.7.1/lib/ruby_parser_extras.rb:1036:in `block in process'
~/.rbenv/versions/2.2.2/lib/ruby/2.2.0/timeout.rb:89:in `block in timeout'
~/.rbenv/versions/2.2.2/lib/ruby/2.2.0/timeout.rb:34:in `block in catch'
~/.rbenv/versions/2.2.2/lib/ruby/2.2.0/timeout.rb:34:in `catch'
~/.rbenv/versions/2.2.2/lib/ruby/2.2.0/timeout.rb:34:in `catch'
~/.rbenv/versions/2.2.2/lib/ruby/2.2.0/timeout.rb:104:in `timeout'
~/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/ruby_parser-3.7.1/lib/ruby_parser_extras.rb:1024:in `process'
~/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/gettext_i18n_rails-1.2.3/lib/gettext_i18n_rails/ruby_gettext_extractor.rb:44:in `run'
~/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/gettext_i18n_rails-1.2.3/lib/gettext_i18n_rails/ruby_gettext_extractor.rb:15:in `parse_string'
~/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/gettext_i18n_rails-1.2.3/lib/gettext_i18n_rails/base_parser.rb:12:in `parse'
~/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/gettext-3.1.4/lib/gettext/tools/xgettext.rb:365:in `block in parse_path'
~/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/gettext-3.1.4/lib/gettext/tools/xgettext.rb:360:in `each'
~/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/gettext-3.1.4/lib/gettext/tools/xgettext.rb:360:in `parse_path'
~/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/gettext-3.1.4/lib/gettext/tools/xgettext.rb:168:in `block in parse'
~/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/gettext-3.1.4/lib/gettext/tools/xgettext.rb:166:in `each'
~/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/gettext-3.1.4/lib/gettext/tools/xgettext.rb:166:in `parse'
~/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/gettext-3.1.4/lib/gettext/tools/xgettext.rb:217:in `generate_pot'
~/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/gettext-3.1.4/lib/gettext/tools/xgettext.rb:151:in `run'
~/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/gettext-3.1.4/lib/gettext/tools/xgettext.rb:34:in `run'
~/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/gettext-3.1.4/lib/gettext/tools/task.rb:330:in `xgettext'
~/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/gettext-3.1.4/lib/gettext/tools/task.rb:314:in `create_pot'
~/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/gettext-3.1.4/lib/gettext/tools/task.rb:306:in `block in define_pot_file_task'
...
```

I don't think there's a syntax problem with the `_db_info.html.haml` file:

https://raw.githubusercontent.com/ManageIQ/manageiq/master/app/views/ops/_db_info.html.haml

and it also makes sense to me to use `Ruby22Parser` for ruby 2.2.

I'm not though entirely sure about the `new_call()` routine from `ruby_gettext_extractor.rb` --
this routine obviously overrides `new_call` from `ruby_parser` gem. It seems that the overriding
version allows for the `args` argument to be `nil`, while later down the road, the code (`extract_key`
function) assumes the value is non-nil.

I don't have a very good insight into this, so -- any hints or advice?

**Comment by grosser** [comment]
Not really ... try reducing it to just this 1 file (remove all other files or make a new testcase here)
and then dive deeper into what triggers the issue ... I'd guess either there is a weird call that does not have arguments or parser api changed

**Comment by mzazrivec** [comment]
I'm closing this PR in favor of pull request #142 which addresses this very issue
in a better way.

**Pull request closed without merging**